docs/concepts/dataclasses.md

```diff
@@ -25,26 +25,26 @@ User(id=42, name='John Doe', signup_ts=datetime.datetime(2032, 6, 21, 12, 0))
 ```
 
 !!! note
-    Keep in mind that `pydantic.dataclasses.dataclass` is **not** a replacement for `pydantic.BaseModel`.
-    `pydantic.dataclasses.dataclass` provides a similar functionality to `dataclasses.dataclass` with the addition of
-    Pydantic validation.
-    There are cases where subclassing `pydantic.BaseModel` is the better choice.
+    Keep in mind that Pydantic dataclasses are **not** a replacement for [Pydantic models](../concepts/models.md).
+    They provide a similar functionality to stdlib dataclasses with the addition of Pydantic validation.
+
+    There are cases where subclassing using Pydantic models is the better choice.
 
     For more information and discussion see
     [pydantic/pydantic#710](https://github.com/pydantic/pydantic/issues/710).
 
-Some differences between Pydantic dataclasses and `BaseModel` include:
+Some differences between Pydantic dataclasses and [`BaseModel`][pydantic.BaseModel] include:
 
-*  How [initialization hooks](#initialization-hooks) work
-*  [JSON dumping](#json-dumping)
+*  [validators](#validators-and-initialization-hooks).
+*  The behavior with the [`extra`][pydantic.ConfigDict.extra] configuration value.
 
 You can use all the standard Pydantic field types. Note, however, that arguments passed to constructor will be copied in
 order to perform validation and, where necessary coercion.
```

Comment:
This is confusing, which _constructor_ are we talking about?

Reply:
This is confusing, but the answer is this happens for both